### PR TITLE
front: editor: fix crash when user change infra while editing switch

### LINKED ISSUE #4801: Front: crash if the user change the current infra while editing a switch
### What happened?

I picked an infra, then click on a switch, and then click on the edition tool

Then I changed the infra.

Crash because of a type error:

Capture d’écran 2023-08-10 à 18 10 35

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Go to Infra Editor
Load an infra
Select a switch
Select Edition Tool
Load another Infra

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

front 1.0.12

### PR DESCRIPTION
**Pull request by kmer2016**
[fix](https://github.com/osrd-project/osrd/issues/4801)

### CODE REVIEW

File: front/src/applications/editor/tools/switchEdition/components.tsx [hunk 1]
```diff
@@ -207,14 +206,23 @@ export const SwitchEditionLeftPanel: FC = () => {
...
+    () => switchType && baseSchema && getSwitchTypeJSONSchema(baseSchema, switchType),
     [baseSchema, switchType]
   );
 
+  /**
+   * FIX crash if the user change the current infra while editing a switch
+   * https://github.com/osrd-project/osrd/issues/4801
+   * No need to display the component if there no switches to display.
+   */
+  if (!switchTypes || switchTypes.length === 0) {
```
Comment (clarani) [inline front/src/applications/editor/tools/switchEdition/components.tsx]: Fix switchType by removing the cast in SwitchType and rather check that switchType is not undefined or null
```suggestion
  if (!switchType || !flatSwitchEntity) {
```

File: front/src/applications/editor/tools/switchEdition/components.tsx [hunk 2]
```diff
@@ -207,14 +206,23 @@ export const SwitchEditionLeftPanel: FC = () => {
...
-    () => baseSchema && getSwitchTypeJSONSchema(baseSchema, switchType),
+    () => switchType && baseSchema && getSwitchTypeJSONSchema(baseSchema, switchType),
     [baseSchema, switchType]
   );
 
+  /**
+   * FIX crash if the user change the current infra while editing a switch
+   * https://github.com/osrd-project/osrd/issues/4801
+   * No need to display the component if there no switches to display.
+   */
```
Comment (clarani) [inline front/src/applications/editor/tools/switchEdition/components.tsx]: I think you can remove the comment
```suggestion

```

**Review by clarani — CHANGES_REQUESTED** [review]

File: front/src/applications/editor/tools/switchEdition/components.tsx (hunk 1 above)
Comment (kmer2016) [reply front/src/applications/editor/tools/switchEdition/components.tsx]: yes, more clean

**Review by clarani — COMMENTED** [review]
tested in local :heavy_check_mark:

**Review by clarani — APPROVED** [review]

**Pull request merged**